lib/senders/index.js

```diff
@@ -23,143 +23,229 @@ module.exports = function (log, config, translator, sender) {
-            redirectTo: opts.redirectTo,
-            resume: opts.resume,
-            acceptLanguage: opts.acceptLanguage || defaultLanguage,
-            ip: opts.ip,
-            location: opts.location,
-            timeZone: opts.timeZone,
-            uaBrowser: opts.uaBrowser,
-            uaBrowserVersion: opts.uaBrowserVersion,
-            uaOS: opts.uaOS,
-            uaOSVersion: opts.uaOSVersion
-          }
-        ))
-      }
-      mailer.sendPasswordChangedNotification = function (email, opts) {
-        return P.resolve(mailer.passwordChangedEmail(
-          {
-            email: email,
-            acceptLanguage: opts.acceptLanguage || defaultLanguage,
-            ip: opts.ip,
-            location: opts.location,
-            uaBrowser: opts.uaBrowser,
-            uaBrowserVersion: opts.uaBrowserVersion,
-            uaOS: opts.uaOS,
-            uaOSVersion: opts.uaOSVersion
-          }
-        ))
-      }
-      mailer.sendPasswordResetNotification = function (email, opts) {
-        return P.resolve(mailer.passwordResetEmail(
-          {
-            email: email,
-            acceptLanguage: opts.acceptLanguage || defaultLanguage,
-            flowId: opts.flowId,
-            flowBeginTime: opts.flowBeginTime,
-          }
-        ))
-      }
-      mailer.sendNewDeviceLoginNotification = function (email, opts) {
-        return P.resolve(mailer.newDeviceLoginEmail(
-          {
-            acceptLanguage: opts.acceptLanguage || defaultLanguage,
-            flowId: opts.flowId,
-            flowBeginTime: opts.flowBeginTime,
-            email: email,
-            ip: opts.ip,
-            location: opts.location,
-            timeZone: opts.timeZone,
-            uaBrowser: opts.uaBrowser,
-            uaBrowserVersion: opts.uaBrowserVersion,
-            uaOS: opts.uaOS,
-            uaOSVersion: opts.uaOSVersion
-          }
-        ))
-      }
-      mailer.sendPostVerifyEmail = function (email, opts) {
-        return P.resolve(mailer.postVerifyEmail(
-          {
-            email: email,
-            acceptLanguage: opts.acceptLanguage || defaultLanguage
-          }
-        ))
-      }
-      mailer.sendUnblockCode = function (account, unblockCode, opts) {
-        return P.resolve(mailer.unblockCodeEmail(
-          {
-            acceptLanguage: opts.acceptLanguage || defaultLanguage,
-            flowId: opts.flowId,
-            flowBeginTime: opts.flowBeginTime,
-            email: account.email,
-            ip: opts.ip,
-            location: opts.location,
-            timeZone: opts.timeZone,
-            uaBrowser: opts.uaBrowser,
-            uaBrowserVersion: opts.uaBrowserVersion,
-            uaOS: opts.uaOS,
-            uaOSVersion: opts.uaOSVersion,
-            uid: account.uid.toString('hex'),
-            unblockCode: unblockCode
-          }
-        ))
-      }
-      return senders
+  .then(function (senders) {
+    var ungatedMailer = senders.email
```

Comment:
Might be overkill, but seems like the logic below is large enough to pull into its own module, `gatedMailer` or something similar, that takes `senders, config, db` and return the gated mailer or the noop mailer.